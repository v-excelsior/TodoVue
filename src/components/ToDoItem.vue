<template>
  <v-list-item class="todolist__item" :class="{completed:todo.isCompleted}" :id="todo.id">
    <v-row justify="center" align="center">
      <v-checkbox v-model="todo.isCompleted" @click="$emit('complete',todo.id)"></v-checkbox>
      <v-col class="text-left">
        <span class='todo-text'>{{todo.text}}</span>
      </v-col>
      <v-col cols="auto">
        <v-btn depressed small color="primary" @click="$emit('edit',todo.id)">Edit</v-btn>
      </v-col>
      <v-col cols="auto">
        <v-btn depressed small color="primary" @click="$emit('delete',todo.id)">Del</v-btn>
      </v-col>
    </v-row>
  </v-list-item>
</template>
<script>
export default {
  name: "ToDoItem",
  props: {
    todo: {
      type: Object,
      require: true
    }
  }
};
</script>

<style lang='scss'>
.completed {
  filter:opacity(0.7);
  span.todo-text{
    text-decoration: line-through;
  }
}
</style>