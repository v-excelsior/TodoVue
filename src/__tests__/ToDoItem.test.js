import { mount } from '@vue/test-utils'
import ToDoItem from '../components/ToDoItem.vue'

import Vue from 'vue'
import Vuetify from 'vuetify';
Vue.use(Vuetify)

const todo = {
    id: 1,
    isCompleted: true,
    text: "Some text",
}

describe('ToDoItem', () => {
    let item
    beforeEach(() => {
        item = mount(ToDoItem, {
            propsData: { todo },
        })
    });

    it('render correctly', () => {
        expect(item.element).toMatchSnapshot()
    })

    it('State checkbox emit state changing', () => {
        let checkbox = item.find('.checkbox input')
        checkbox.trigger('change')
        expect(item.emitted().toogleState).toBeTruthy()
    })

    it('Delete button emit delete handler', () => {
        let delBtn = item.find('.del-btn')
        delBtn.trigger('click')
        expect(item.emitted().delTodo).toBeTruthy()
    })

    it('Edit button emit edit handler', () => {
        let delBtn = item.find('.edit-btn')
        delBtn.trigger('click')
        expect(item.emitted().editTodo).toBeTruthy()
    })
})
