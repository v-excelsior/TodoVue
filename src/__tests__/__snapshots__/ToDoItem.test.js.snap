// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoItem render correctly 1`] = `
<div>
  <div
    class="ml-3 list-item v-list-item theme--light"
    id="1"
    tabindex="-1"
  >
    <div
      class="v-input checkbox v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--checkbox primary--text"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input--selection-controls__input"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate material-icons theme--light primary--text"
            >
              $checkboxOn
            </i>
            <input
              aria-checked="true"
              id="input-3"
              role="checkbox"
              type="checkbox"
              value=""
            />
            <div
              class="v-input--selection-controls__ripple primary--text"
            />
          </div>
        </div>
        <div
          class="v-messages theme--light primary--text"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
     
    <div
      class="v-list-item__content"
    >
      <div
        class="v-list-item__subtitle todo-text ma-0 completed"
      >
        Some text
      </div>
    </div>
     
    <div
      class="v-list-item__action"
    >
      <button
        class="edit-btn v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--small primary--text"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-pencil theme--light"
          />
        </span>
      </button>
    </div>
     
    <div
      class="v-list-item__action"
    >
      <button
        class="del-btn v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--small primary--text"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-delete theme--light"
          />
        </span>
      </button>
    </div>
  </div>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--dark"
    role="separator"
  />
</div>
`;
